### PR TITLE
Feat/RSS-ECOMM-3 display product variants

### LINKED ISSUE #144: Issue RSS-ECOMM-3_no-issue-display variants if any

### PR DESCRIPTION
**Pull request by Tetiana-KET**
#### Feat/RSS-ECOMM-3 display product variants

resolve #144 

#### 🖊️ Description
display product variants to user
display corresponding photo and descrioption, when variant is selected

#### 📝 Additional Information

**Screenshots/Links:**
image

#### Checklist

- [x] ✅ I have performed a self-review of my own code.
- [x] 📝 I have commented my code, particularly in hard-to-understand areas.
- [x] 🔧 I have made corresponding changes to the documentation (if applicable).
- [x] 🚫 My changes generate no new warnings or errors.
- [x] 👌 Have at least 1 approve.

### CODE REVIEW

File: src/components/Counter/Counter.component.tsx [hunk 1]
```diff
@@ -37,7 +37,12 @@ function CounterComponent({ onChange, initCount = minCount }: CounterComponentPr
   };
 
   return (
-    <Stack direction="row" alignItems="center" gap="2px">
+    <Stack
+      direction="row"
+      alignItems="center"
+      gap="2px"
+      sx={{ position: 'absolute', left: '50%', transform: 'translateX(-50%)' }}
```
Comment (Gromov-Roman) [inline src/components/Counter/Counter.component.tsx]: this is not good idea, please add sx input prop and provide it in ProductCard component

File: src/pages/Product/Product.page.tsx [hunk 2]
```diff
@@ -17,12 +19,26 @@ function ProductPage() {
...
+    if (data?.masterVariant) {
+      setSelectedVariant(data.masterVariant);
+    }
+  }, [data]);
+
+  const handleVariantClick = (variant: ProductVariant) => {
+    setSelectedVariant(variant);
+  };
+
+  const getColorAttribute = (variant: ProductVariant) =>
```
Comment (JsPowWow) [inline src/pages/Product/Product.page.tsx]: It seems might me used outside of component as pure function; Move it to the some utils

File: src/pages/Product/Product.page.tsx [hunk 3]
```diff
@@ -17,12 +19,26 @@ function ProductPage() {
...
+    variant.attributes?.find((attr) => attr.name === 'color')?.value || 'primary';
 
   const productInfo = generateProductObj(data);
 
   const defaultImageUrl = '/public/defaultImg.png';
```
Comment (JsPowWow) [inline src/pages/Product/Product.page.tsx]: Why this is here ? move it to the appropriate constants file

File: src/pages/Product/Product.page.tsx [hunk 4]
```diff
@@ -17,12 +19,26 @@ function ProductPage() {
...
-  const images = data?.masterVariant.images
-    ? [...data.masterVariant.images].map((image) => ({
+  const images = selectedVariant?.images
```
Comment (JsPowWow) [inline src/pages/Product/Product.page.tsx]: It seems might be used outside of component as pure function; Move it to the some utils

File: src/pages/Product/Product.page.tsx (hunk 2 above)
Comment (Tetiana-KET) [reply src/pages/Product/Product.page.tsx]: done

File: src/pages/Product/Product.page.tsx (hunk 3 above)
Comment (Tetiana-KET) [reply src/pages/Product/Product.page.tsx]: done

File: src/pages/Product/Product.page.tsx (hunk 4 above)
Comment (Tetiana-KET) [reply src/pages/Product/Product.page.tsx]: thank you for your remarks, done.

**Review by Gromov-Roman — APPROVED** [review]

**Pull request merged**